huntsman/dome/musca.py

```diff
@@ -120,10 +123,35 @@ def open(self):
 
         v = self.status()[Protocol.SHUTTER]
         if v == Protocol.OPEN:
+            # refresh the threading close event
+            self._close_event.clear()
+            self.logger.info('Starting thread to keep dome open.')
+            # start a thread to send the keep dome open command to musca
+            keep_open = threading.Thread(target=self.keep_dome_open)
             return True
         self.logger.warning('HuntsmanDome.open wrong final state: {!r}', v)
         return False
 
+    def keep_dome_open(self):
+        """Periodically tell musca to reset watchdog timer
+
+        """
+        # maximum number of loops before dome closure (correspond to 15hrs)
+        max_loopps = 186
+        for i in range(max_loops):
+            # check to see if a dome closure has occured
+            if self._close_event().is_set():
+                self.logger.info('Keep dome open thread has detected a dome closure, ending thread.')
+                # if dome has closed, don't try to 'keep dome open'
+                break
```

Comment:
Now that the enclosing loop is `for` then this probably wants to go back to being a `return`... Otherwise you'll get that warning about maximum loops every time you close the dome normally.